src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1719,6 +1720,29 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
         this->resetACTNUM(actnum.data());
     }
 
+    void EclipseGrid::
+    updateACTNUMWithAQUNUM(const Deck& deck,
+                           std::vector<int>& actnum_vector) const {
+        using AQUNUM =ParserKeywords::AQUNUM;
+        if ( !deck.hasKeyword<AQUNUM>() ) {
+            return;
+        }
+        const auto& aqunum_keywords = deck.getKeywordList<AQUNUM>();
+        for (const auto& keyword : aqunum_keywords) {
+            for (const auto &record : *keyword) {
+                const size_t i = record.getItem<AQUNUM::I>().get<int>(0) - 1;
+                const size_t j = record.getItem<AQUNUM::J>().get<int>(0) - 1;
+                const size_t k = record.getItem<AQUNUM::K>().get<int>(0) - 1;
+                const size_t global_index = this->getGlobalIndex(i, j, k);
+                if (actnum_vector[global_index] == 0) {
```

Comment:
Actually - I have come to the conclusion that this should be "opposite"; the cell *should* have `ACTNUM == 0`- and we should print a warning if `ACTNUM == 1`.

Reply:
I understand what you mean here. I felt so weird to see how the numerical aquifer is implemented, that some reservoir cells are specified to be aquifer cells, which is not convenient. But now I understand probably they just use inactive cells to do it. But from the syntax side, there is no requirement that they should be specified with inactive cells. We have cases both with active cells and inactive cells, we do not see any warning/information related to them.